### PR TITLE
Install mod from branch

### PR DESCRIPTION
**Pull request by tylerklose**
## What does this PR do?
Provides a way to install a mod from a specific branch of a git repo by providing the branch name as a flag `--branch branch-name`

Rather than taking in `--github user/repo/path/to/mod` or `--gitlab user/repo/path/to/mod`, accept `--url https://github.com/user/repo --path path/to/mod`

~Also makes use of the dynamically generated parameter accessors from https://github.com/gotta-dev/gotta-cli/pull/10~

## Do you have any concerns about this body of work? Anything you'd like to highlight or open for discussion? Anything to be aware of before merging / deploying?

~Depends on https://github.com/gotta-dev/gotta-cli/pull/10~

## Where should the review start?
`lib/gotta/mod/repository.rb`

## Recommendation/Steps for how to test this
`gotta install mod webpack --url https://github.com/gotta-dev/mods --path webpack --branch stimulus`

## Link to the related issue(s)
~https://github.com/gotta-dev/gotta-cli/pull/10~

### CODE REVIEW

**Comment by tylerklose** [comment]
@parruda this no longer depends on https://github.com/gotta-dev/gotta-cli/pull/10. Ready for review

**Comment by parruda** [comment]
@tylerklose
Using `--url` could be a bit confusing. If it is a URL, why do I have to specify a path and a branch?

As a user, the way I'd expect to use it like this:
```
gotta install mod webpack --git https://github.com/gotta-dev/mods
# OR
gotta install mod webpack --git git@github.com:gotta-dev/mods
# OR
gotta install mod webpack --github|--gitlab gotta-dev/mods
```

I think the following are already covered, but I am listing here for documentation:
`--path` defaults to the root of the repo if not specified
`--branch` defaults to `master` if not specified

File: lib/gotta/mod/cli/common.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        def mod_name
+          parameter 'mod_name'
+        end
+
+        def branch
+          option '--branch'
+        end
+
+        def host
```
Comment (parruda) [inline lib/gotta/mod/cli/common.rb]: What's `--host` for?

**Comment by parruda** [comment]
@tylerklose Would you mind adding some tests to the files you touched? That'd be a great way to slowly bring the coverage up.

File: lib/gotta/mod/cli/common.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      class CommonMod < Gotta::CLI::Command::Base
+        private
+
+        def mod_name
+          parameter 'mod_name'
+        end
+
+        def branch
+          option '--branch'
```
Comment (parruda) [inline lib/gotta/mod/cli/common.rb]: I'd add here a default.

`option '--branch', default: 'master'`

That way we can display defaults when we add the autogenerated help.

File: lib/gotta/mod/cli/common.rb (hunk 1 above)
Comment (tylerklose) [reply lib/gotta/mod/cli/common.rb]: That's left over, good catch. I was thinking about doing `--host github` or `--host gitlab` instead, but I think the URL is cleaner

**Comment by tylerklose** [comment]
I will change `--url` to `--git` and add defaults for `--path` and `--branch`. I will also add some test cases as well

**Comment by tylerklose** [comment]
The reason I removed the ability to do `gotta install mod webpack --github gotta-dev/mods` / `gotta install mod webpack --github gotta-dev/mods` is because the code was pretty weird to unpack it and it would only get worse if we wanted to support `bitbucket` and other hosts. Is it desirable to have it?

File: lib/gotta/mod/cli/common.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        def repo_parameters
+          { mod_name: mod_name, git: git, branch: branch, path: path }
+        end
+
+        def branch
+          option '--branch', default: 'master'
+        end
+
+        def path
+          option '--path', default: mod_name
```
Comment (parruda) [inline lib/gotta/mod/cli/common.rb]: I like that a lot!

**Pull request merged**